### ISSUE TITLE
Python 3.9.0 and 3.9.1.rc1++

### ISSUE THREAD

**Issue by hasii2011** [ronaldoussoren/py2app#320]
I am successful at packaging on Python 3.8.5 (Pillow 6.2.2).   However, on both the above I get this error at program launch,

Pillow is version 8.0.0 for Python 3.9.x (anything less won't compile)

ERROR PyutApp: dlopen(/Users/humberto.a.sanchez.ii/PycharmProjects/PyUt/dist/Pyut.app/Contents/Resources/lib/python3.9/lib-dynload/PIL/_imaging.so, 2): Library not loaded: @loader_path/.dylibs/libxcb.1.1.0.dylib
  Referenced from: /Users/humberto.a.sanchez.ii/PycharmProjects/PyUt/dist/Pyut.app/Contents/Resources/lib/python3.9/lib-dynload/PIL/_imaging.so
  Reason: image not found

--------------------
Is this a 3.9.x issue, a recipe issue, or something else?  BTW, I am also on Mac OS Big Sur;

**Comment by stephencwest** [comment]
I'm having the same problem with psycopg2-binary on python3.9 trying to load @loader_path/.dylibs/libpq.5.11.dylib.  I can see the dylib is in the Frameworks directory of the bundle.

**Comment by ronaldoussoren** [comment]
I can reproduce the problem with Pillow (and haven't tried to reproduce this using psycopg2 as it likely has the same root cause).

A quick workaround is to force the inclusion of the entire package, that is add the following to setup.py:

```
setup(
    ...
    options=dict(py2app=dict(packages="PIL"))
```

For psycopg2 use its python package instead of PIL.

I'm looking into a fix for this in py2app itself.

**Comment by ronaldoussoren** [comment]
Could you check if the version of py2app in the git repo fixes this problem for you? 

The commit earlier in the discussion thread adds generic support for handling "@loader_path" linker commands to py2app. I've checked that this fixes the issue for the PIL example in the py2app repository.

My current plan is to push out a release of py2app with this fix before the new year.

**Comment by ronaldoussoren** [comment]
This should be fixed in py2app 0.23 (just released)
